### PR TITLE
Make time format configruable.

### PR DESCRIPTION
**Pull request by VihasMakwana**
- Issue: https://github.com/signalfx/splunk-otel-collector-chart/issues/644
- If the host is not set to UTC, the current logic breaks. To resolve this issue, the user needs to override current operators in `agent.config` section, and it wouldn't look so good.
- Allow users to modify the timestamp string in the filelog receiver. It uses current values by default, but give the option to override them through `logsCollection.containers.timestampExtraction`

### CODE REVIEW

**Comment by jvoravong** [comment]
Hey @VihasMakwana,
Can you describe in more detail what actually breaks or doesn't work as expected? Providing example logs and results would also be very helpful.

File: helm-charts/splunk-otel-collector/values.yaml [hunk 1]
```diff
@@ -575,6 +575,17 @@ logsCollection:
     # Set to 0 to remove any size limit.
     maxRecombineLogSize: 1048576
 
+    # Regex to extract logs, time for CRI-O or containerd logs
+    # More info: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/operators/regex_parser.md
+    regexp: ""
```
Comment (jvoravong) [inline helm-charts/splunk-otel-collector/values.yaml:580]: A Kubernetes environment can use different container runtimes on different nodes at the same time. The suggested override configurations should be granular enough to configure each parser (crio, containerd, docker) individually.

Comment (VihasMakwana) [reply helm-charts/splunk-otel-collector/values.yaml:580]: Okayy, will take care of this one and update the PR

**Comment by VihasMakwana** [comment]
> Hey @VihasMakwana, Can you describe in more detail what actually breaks or doesn't work as expected? Providing example logs and results would also be very helpful.


From the linked issue:

- **Our logs pipeline assumes kubernetes host will be in UTC. While this is definitely the most common scenario, and we do encourage users to always use UTC, if the host is not set to UTC, it breaks our hardcoded parsing logic.**
- **Ideally we should improve the regex to be able to handle the chance the host timestamp may not be UTC/Z and parse out the timestamp correctly.**

Example of contained runtime with a different timezone:
```
2023-01-23T12:09:28.344779078-06:00 stdout F {"caller":"service_controller.go:103","controller":"ServiceReconciler","end reconcile":"kube-system/microk8s.io-hostpath","level":"info","ts":"2023-01-23T18:09:28Z"}
2023-01-23T12:09:28.344779078-06:00 stdout F {"caller":"service_controller.go:103","controller":"ServiceReconciler","end reconcile":"kube-system/microk8s.io-hostpath","level":"info","ts":"2023-01-23T18:09:28Z"}
```
The above wouldn't match the hardcoded regex.

For more details about reproducing this, refer https://github.com/signalfx/splunk-otel-collector-chart/issues/644#issuecomment-1589344762.

**Comment by dmitryax** [comment]
Is the issue fixed now by https://github.com/signalfx/splunk-otel-collector-chart/pull/817 ? I would like to avoid introducing new configuration options if they are not strictly required

**Comment by jvoravong** [comment]
Left a similar [comment](https://github.com/signalfx/splunk-otel-collector-chart/issues/644#issuecomment-1601143341) as @dmitryax in the related issue. Was waiting for confirmation from involved reporters.

**Comment by VihasMakwana** [comment]
Closing this one, will raise another PR for https://github.com/signalfx/splunk-otel-collector-chart/issues/829

**Pull request closed without merging**